dataloader.py

```diff
@@ -50,6 +50,34 @@ def __getitem__(self, ind):
         
         #Return features/labels/lengths_of_game as a dictionary
         return {"features":torch.Tensor(features), "labels":torch.Tensor(labels), "lengths":length}
+    
+    
+class Hero2vecDataset(Dataset):
+    '''
+    The dataloader class to load data for the hero2vec training.
+    '''
+    
+    def __init__(self, lineup_file = './data/pro_lineup.txt'):
+        assert os.path.exists(lineup_file), "Lineup file doesn't exist."
+        self.lineup = np.loadtxt(lineup_file) - 1
+        
+    def __len__(self):
+        return len(self.lineup) * 5
+    
+    def __getitem__(self, ind):
+        '''
+        Inputs:
+            ind: int
+        Returns:
+            dictionary:
+                        "context": torch.LongTensor, size = (4,). The 4 context heroes within the team.
+                        "target": int. The target hero
+        '''
+        game_ind, hero_ind = divmod(ind,5)
+        context_heroes = np.concatenate((self.lineup[game_ind, :hero_ind], self.lineup[game_ind, hero_ind+1:]))
+        target = self.lineup[game_ind, hero_ind]
+
+        return {"context":torch.LongTensor(context_heroes), "target":int(target)}
```

Comment:
Well written class!

This isn't necessary for this PR, but make a note to yourself to go back and add docstrings wherever you can after week 4 is over.